### ISSUE TITLE
tests in firefox 57 the nvda stop once and for all when getting lost in reading the pages

### ISSUE THREAD

**Issue by fernando-jose-silva** [nvaccess/nvda#7633]
I installed firefox 64bits 57 that is in tests, more of curiosity to see if nvda worked well, motivated by the engine change that required changes from other screen readers.
When reading a few pages, when walking with the arrows up and down, when the focus goes through an advertising area the nvda looks starts reading the first line, soon interrupts the reading:
Open the page
https://tecnoblog.net/213514/traduzir-pagina-web-ingles-firefox/
Read the open web page with arrows down to the line
PUBLICIDADE
by pressing the next arrow below see that nvda
read  fre
beginning of the word frame
the expected would be that the nvda reads:
clickable frame
see that while continuing to read the page with arrow below the nvda fails to read the next line
should pronounce
clickable
When pressing one more arrow below the nvda should inform the output of the frame, however nvda repeats the title of the page and does not read the contents of the first line after the exit of the frame:
inRead invented by Teads
I persisted in this behavior on other sites as well.
In some sites when navigating with the arrows, nvda also repeats the last line previously spoken and suppresses the current line, when I find these cases again I include the link.
Thank you.
using nvda next 14463 firefox 64 bits version 57, and windows 7 64bits

**Comment by leonardder** [comment]
I can't find the word PUBLICIDADE on this page.

**Comment by Brian1Gaff** [comment]
I've not tried this, but when I mentioned the engine change in the dev list, 
nobody seemed to know what i was on about.
 This would explain why they have not back ported the fix for the focus 
issue back to 56.
 I personally always have issues with 64 bit Firefox an nvda, who is why I'm 
using the 32 bit version it would be interesting to try 57 32 bit though, 
but not had the time yet.
 Could the word being not there be due to the language of the page being 
different?
 I'd also be interested to know if  the ad blockers work in this new version 
of firefox.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
----- Original Message -----

**Comment by fernando-jose-silva** [comment]
Strange, perhaps advertising depending on the region where you are accessing, you look for other example, and envy.
Enviado do meu iPhone

Em 28 de set de 2017, à(s) 01:57, Leonard de Ruijter <notifications@github.com> escreveu:

**Comment by fernando-jose-silva** [comment]
Another place where you can test, open the nv access page itself
Walk with the arrows down until shortly after the frame
NV Access: Our Mission (Developers of free screen reader NVDA)
nvda will read the line:
NVDA is open source software. If you would like to contribute to the project, visit
Now press an arrow below and nvda should read the link:
NVDA on GitHub.
However nvda repeats the title of the page.
You can now walk with arrows
Until you get to the frame again, as soon as you enter the frame the nvda should read:
Play button
But notice that the nvda starts reading a word or phrase, and is silent not informing the button in focus.
Try to navigate with the arrows without jumping directly to the items.
Also try to always navigate with the arrows until all the items are located, for example the Play button is read when navigating with the arrows below, but is not read when navigating with several arrows below and returns with arrows up.



You can try another test on the site:
http://blogs.opera.com/news/
As soon as you enter the page you can press ctrl + home.
Note that nvda no longer has text in focus.
Now walk with arrows below:
Will go through the items:
Download Now
one more arrow below
clickable
one more arrow below
Note that nvda will now repeat the title of the page.
But the link that is in focus is:
Opera blogs
The nvda should inform:
graphic link Opera blogs
If you press an arrow below and return with an aseptic arrow you will see that the correct information will be read.
I hope you can see.
Otherwise, try installing firefox 57 64bits and use it for a few hours by browsing several websites that will find several places with this same behavior.

**Comment by Brian1Gaff** [comment]
As I say I had to downdate from 56 due to focus issues which they claimed to 
have fixed in 57, kind of looks like in fixing one thing they screwed up 
something else.
 Oh dear.


bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.

**Comment by Brian1Gaff** [comment]
I think you need to explain this in a bugzilla report as they might 
understand what has gone wrong. Needs somebody to test 57 32 bit to see if 
its the same. When it comes along I can try it , but only on windows 7.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
-----

**Comment by tapper82** [comment]
Hi I have had brackage on this page to:
https://people.xiph.org/~jm/demo/rnnoise/donate.html
Buttens wont press. Windows 10 ff 58 64 bit

**Comment by dan1982code** [comment]
Not sure whether to open new issues for Firefox 57, being as it's in beta, but I've noticed a few things with NVDA and Firefox 57 beta 5:

1. Loading pages is sometimes much much slower. For example, open the NVDA What's New document. On Firefox 56, it takes under a second; in Firefox 57, it takes 5-10 seconds.

2. NVDA now says "loading document..." a lot. I suppose that this is for big pages?

3. Hitting ctrl+j doesn't focus the downloads as expected; instead, you have to tab and then shift+tab to get to the downloads.

Could the OP explain the "focus issue" in Firefox 56? I haven't had any focus trouble in Firefox 56 so far.

**Comment by Brian1Gaff** [comment]
In 56 it goes to the sidebar I remember raising it on bugzilla and they said 
it was not portable back to 56 but was fixed in 57. When I mentioned here 
the massive changes in 57, nobody seemed to know what I was talking about!
 I personally think they have messed it up big time, as its engine is 
renamed to photon or some other word that sounds like it not the original 
one its kind of like son of Frankenstein!
 I'm staying on 55 32 bit till they get their act sorted out. Now that Jamie 
is over there maybe somebody could ask him.
 grin.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
-----

**Comment by RichCaloggero** [comment]
## Same behavior with 57 beta 5, 32-bit

Let me try and summarize the behavior again.

- got http://nvaccess.org
- top of page with control+home
- press "m" to goto first frame
	+ expected: "frame clickable clickable"
	+ behavior: starts to say something, then speech is immediately truncated
- press control+home to goto start of page again
	+ expected: read first line of page ( "heading level 1, NV Access: Empowering lives through non-visual access to technology" )
	+ behavior: reads the window's title bar ( something like "nvAccess - mozilla firefox" )

The bug is triggered only when you move in-and-out of iFrames. So, if we do same as above, but from the bottom of the video player frame on nvAccess, we get similar behavior.

This occurs in 57.0-beta5 and firefox developer edition, also based on 57.0-beta5. I did not try developer edition 32-bit, but the 57.0-beta5 32-bit version exhibits same buggy behavior.

Hope this is helpful.

**Comment by jcsteh** [comment]
This is probably [Mozilla bug 1415177](https://bugzilla.mozilla.org/show_bug.cgi?id=1415177). Unfortunately, this is difficult to investigate because it can't be reliably reproduced. On some systems, it occurs sometimes. On others, it occurs more. On some, it doesn't occur at all. That said, I was seeing this a few times a week, but after a lot of performance fixes and cleanup landed, I haven't seen this for weeks. So, I'm hoping it's fixed for Firefox 58 or 59, but I'm keeping an eye on it.

**Comment by Brian1Gaff** [comment]
I wonder as somebody who has regressed away from 57 due to it being often 
unresponsive and sometimes not working at all in my windows 7 system, does 
there exist a version of Firefox 58 or up that can be run portably for 
testing or is this not possible. It would be nice to be able to boot it up 
at will on the same machine as the ESR lives on to see how its coming along.

Brian

**Comment by jcsteh** [comment]
Can anyone still reproduce this in Firefox 58 beta? I'm fairly certain this has been fixed by recent changes, as I haven't seen it in months, but it'd be good to know which release fixes it: 58 (currently beta) or 59 (nightly).

**Comment by jcsteh** [comment]
[**@Brian1Gaff**](https://github.com/Brian1Gaff) commented on [Dec 4, 2017, 8:33 PM GMT+10](https://github.com/nvaccess/nvda/issues/7633#issuecomment-348922702 "2017-12-04T10:33:21Z - Replied by Github Reply Comments"):
> I wonder as somebody who has regressed away from 57 due to it being often 
> unresponsive and sometimes not working at all in my windows 7 system, does 
> there exist a version of Firefox 58 or up that can be run portably for 
> testing or is this not possible.

You can get a zip version, but you would need to register a dll as administrator (regsvr32 AccessibleHandler.dll) and also run it with a different profile (since otherwise it might break the profile you're using with an older version). Here's a [zip archive for Firefox 58.0b15](http://archive.mozilla.org/pub/firefox/candidates/58.0b15-candidates/build1/win64/en-US/firefox-58.0b15.zip).

**Comment by jcsteh** [comment]
Oh, and you can create/switch profiles by passing the `-profilemanager` command line option.

**Comment by Brian1Gaff** [comment]
I have the file now...

When you say run from a different profile, do you mean a different windows 
user, or can one just tweak something to have the two versions  allowed to 
coexist on the same user. The latter would of course be better for 
comparison purposes.
 Brian

**Comment by Brian1Gaff** [comment]
Hi, here is what I have done. its probably not to be recommended but..
     I unzipped the files to the x86 part of program files and just ran the 
firefox.exe.
 That seems to be fairly reliable of course it disables navigational sounds, 
but If I run the code from the correct 64 bit folder it continually asks to 
be the default browser and completely trashes navigational sounds in version 
52, whereas as it stands it merely disables it and  a second run of 52 will 
restore it.
 The only bit I could not get to work this way thus far is download 
dialogue, ie when you open that window it just presents unknown even though 
you know you have downloaded a file.
 I'm a little afraid if I try to install 58 properly, no matter where I 
shove it it will not see the bookmarks or if I let it do so, it will trash 
52 in the process.
 Anyway, 58 seems a lot more reliable. I simply could not get 57 to even 
produce anything but menus  before.
 Ideally, if Firefox is not going to be able to have navigational sounds 
installed, maybe Mozilla might like to make such sounds an option in built. 
It is very handy to know when a download is complete and when a page is 
refreshing itself by listening for the clicks or has finished loading by 
that little honk.
 Brian
